### PR TITLE
feat: improve DX & e2e

### PR DESCRIPTION
**Pull request by Himself65**
enhance continuously debugging using monorepo and swc watch

```shell
# in the first shell
pnpm run dev
```

```shell
# in the second shell
cd ./examples/xxx
waku dev
```

### CODE REVIEW

**Comment by Himself65** [comment]
```
/Users/himself65/.nvm/versions/node/v18.16.0/bin/pnpm run e2e

> waku@0.11.2 e2e /Users/himself65/Code/wakuwork
> playwright test


Running 3 tests using 3 workers

[WebServer] Error when using sourcemap for reporting an error: Can't resolve original location of error.
[WebServer] Module level directives cause errors when bundled, "use client" in "src/Counter.tsx" was ignored.
[WebServer] Error when using sourcemap for reporting an error: Can't resolve original location of error.
[WebServer] Module level directives cause errors when bundled, "use client" in "src/Counter.tsx" was ignored.
[WebServer] Error when using sourcemap for reporting an error: Can't resolve original location of error.
[WebServer] Error when using sourcemap for reporting an error: Can't resolve original location of error.
[WebServer] Module level directives cause errors when bundled, "use client" in "src/Counter.tsx" was ignored.
[WebServer] Error when using sourcemap for reporting an error: Can't resolve original location of error.
[WebServer] Module level directives cause errors when bundled, "use client" in "src/Counter.tsx" was ignored.
[WebServer] Error when using sourcemap for reporting an error: Can't resolve original location of error.
  ✓  1 e2e/counter.spec.ts:4:1 › counter (650ms)
  ✓  2 e2e/counter.spec.ts:13:1 › first render (502ms)
  ✓  3 e2e/async.spec.ts:4:1 › async (1.5s)

  3 passed (2.4s)

Process finished with exit code 0
```

**Comment by Himself65** [comment]
> ```
> /Users/himself65/.nvm/versions/node/v18.16.0/bin/pnpm run e2e
> 
> > waku@0.11.2 e2e /Users/himself65/Code/wakuwork
> 
> > playwright test
> 
> 
> Running 3 tests using 3 workers
> 
> [WebServer] Error when using sourcemap for reporting an error: Can't resolve original location of error.
> [WebServer] Module level directives cause errors when bundled, "use client" in "src/Counter.tsx" was ignored.
> [WebServer] Error when using sourcemap for reporting an error: Can't resolve original location of error.
> [WebServer] Module level directives cause errors when bundled, "use client" in "src/Counter.tsx" was ignored.
> [WebServer] Error when using sourcemap for reporting an error: Can't resolve original location of error.
> [WebServer] Error when using sourcemap for reporting an error: Can't resolve original location of error.
> [WebServer] Module level directives cause errors when bundled, "use client" in "src/Counter.tsx" was ignored.
> [WebServer] Error when using sourcemap for reporting an error: Can't resolve original location of error.
> [WebServer] Module level directives cause errors when bundled, "use client" in "src/Counter.tsx" was ignored.
> [WebServer] Error when using sourcemap for reporting an error: Can't resolve original location of error.
>   ✓  1 e2e/counter.spec.ts:4:1 › counter (650ms)
>   ✓  2 e2e/counter.spec.ts:13:1 › first render (502ms)
>   ✓  3 e2e/async.spec.ts:4:1 › async (1.5s)
> 
>   3 passed (2.4s)
> 
> Process finished with exit code 0
> ```
> 
> Refs: #49

See https://github.com/dai-shi/waku/pull/63

**Comment by dai-shi** [comment]
Is this suboptimal for #64 structure?
Should we merge this now and your follow up later?

**Comment by Himself65** [comment]
I think we should merge it now to avoid this pr too complexw

**Comment by Himself65** [comment]
And I found that this repo doesn't have the eslint rule, which will make extra commits

File: package.json [hunk 1]
```diff
@@ -47,12 +47,13 @@
...
   "scripts": {
+    "dev": "swc src -d dist -w",
     "compile": "npm run compile:code && npm run compile:types",
     "compile:code": "swc src -d dist && swc src -d dist/cjs -C module.type=commonjs && echo '{\"type\":\"commonjs\"}' > dist/cjs/package.json",
     "compile:types": "tsc --project tsconfig.build.json",
     "test": "tsc --project . --noEmit",
-    "e2e": "cd e2e/01 && playwright test && cd ../02 && playwright test",
-    "examples:dev": "npm run compile:code && ./dist/cli.js dev --config ./examples/${NAME}/vite.config.ts",
+    "e2e": "playwright test",
+    "examples:dev": "npm run compile:code && waku dev --config ./examples/${NAME}/vite.config.ts",
```
Comment (dai-shi) [inline package.json:56]: Interesting to see `waku dev` works here. Should we do the same for others?

**Review by dai-shi — APPROVED** [review]
This is still new to me, and I may want to change something in the future, but let's merge it for now.

**Comment by dai-shi** [comment]
> And I found that this repo doesn't have the eslint rule, which will make extra commits

I will work on eslint and prettier. Yeah, it's about the time.

File: package.json (hunk 1 above)
Comment (Himself65) [reply package.json:56]: for sure

**Pull request merged**